### PR TITLE
Add revert error reason at klay_call, klay_estimateGas

### PR DESCRIPTION
**Pull request by hqjang-pepper**
## Proposed changes

- Add revert error reason at revert message when `klay_estimateGas` and `klay_call` is called
- fix gofumpt at `api` package

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: api/api_public_blockchain.go [hunk 1]
```diff
@@ -311,12 +282,26 @@ func DoCall(ctx context.Context, b Backend, args CallArgs, blockNrOrHash rpc.Blo
...
+		return nil, 0, 0, 0, err
+	}
+	// The intrinsicGas is checked again later in the blockchain.ApplyMessage function,
+	// but we check in advance here in order to keep StateTransition.TransactionDb method as unchanged as possible
+	// and to clarify error reason correctly to serve eth namespace APIs.
+	// This case is handled by EthDoEstimateGas function.
+	if msg.Gas() < intrinsicGas {
+		return nil, 0, 0, 0, fmt.Errorf("%w: msg.gas %d, want %d", blockchain.ErrIntrinsicGas, msg.Gas(), intrinsicGas)
+	}
+	evm, vmError, err := b.GetEVM(ctx, msg, st, header, vm.Config{})
```
Comment (jimni1222) [inline api/api_public_blockchain.go:302]: Why did not you use `vmCfg`?

Comment (hqjang-pepper) [reply api/api_public_blockchain.go:302]: Thanks for the review. I'll reflect it.

File: api/api_public_blockchain.go [hunk 2]
```diff
@@ -311,12 +282,26 @@ func DoCall(ctx context.Context, b Backend, args CallArgs, blockNrOrHash rpc.Blo
...
-	state.AddBalance(msg.ValidatedSender(), new(big.Int).Mul(new(big.Int).SetUint64(msg.Gas()), msg.GasPrice()))
-	// Get a new instance of the EVM.
-	evm, vmError, err := b.GetEVM(ctx, msg, state, header, vmCfg)
+	// TODO-Klaytn: Klaytn is using fixed baseFee as now but, if we change this fixed baseFee as dynamic baseFee, we should update this logic too.
+	fixedBaseFee := new(big.Int).SetUint64(params.BaseFee)
+	intrinsicGas, err := types.IntrinsicGas(args.Data, nil, args.To == nil, b.ChainConfig().Rules(header.Number))
+	if err != nil {
+		return nil, 0, 0, 0, err
+	}
+	msg, err := args.ToMessage(globalGasCap.Uint64(), fixedBaseFee, intrinsicGas)
```
Comment (jimni1222) [inline api/api_public_blockchain.go:291]: Can this logic handle the case where the optional values are not defined in args?

File: api/api_public_blockchain.go [hunk 3]
```diff
@@ -380,28 +381,53 @@ func (s *PublicBlockChainAPI) DoEstimateGas(ctx context.Context, b Backend, args
...
-
-		_, _, _, failed, err := DoCall(ctx, b, args, rpc.NewBlockNumberOrHashWithNumber(rpc.LatestBlockNumber), vm.Config{UseOpcodeComputationCost: true}, localTxExecutionTime, gasCap)
-		if err != nil || failed {
-			return false
+		ret, _, _, status, err := DoCall(ctx, b, args, rpc.NewBlockNumberOrHashWithNumber(rpc.LatestBlockNumber), vm.Config{}, 0, gasCap)
+		if err != nil {
+			if errors.Is(err, blockchain.ErrIntrinsicGas) {
+				// Special case, raise gas limit
+				return false, ret, nil, nil
+			}
```
Comment (hyunsooda) [inline api/api_public_blockchain.go:391]: It seems a key change for the `eth_call` in this PR. If this is correct, the unnecessary changes (e.g., changed function signature, etc)  would not be made.

File: api/api_public_blockchain.go [hunk 4]
```diff
@@ -380,28 +381,53 @@ func (s *PublicBlockChainAPI) DoEstimateGas(ctx context.Context, b Backend, args
...
+			// Returns error when it is not VM error (less balance or wrong nonce, etc...).
+			return false, nil, nil, err
```
Comment (hyunsooda) [inline api/api_public_blockchain.go:393]: I think you may consider intrinsics gas only. The gas error can obviously be returned by the runtime of the EVM.

File: api/api_public_blockchain.go [hunk 5]
```diff
@@ -380,28 +381,53 @@ func (s *PublicBlockChainAPI) DoEstimateGas(ctx context.Context, b Backend, args
...
+		if !isExecutable {
+			if vmErr != nil {
+				// Treat vmErr as RevertError only when there was returned data from call.
+				if isReverted(vmErr) && len(ret) > 0 {
+					return 0, newRevertError(ret)
+				}
+				return 0, vmErr
+			}
+			// Otherwise, the specified gas cap is too low
+			return 0, fmt.Errorf("gas required exceeds allowance (%d)", cap)
```
Comment (hyunsooda) [inline api/api_public_blockchain.go:430]: As the comment above (misconsidered runtime gas usage (`constantGas` + `dynamicGas`), this routine seems to be reconsidered. If you filter the class of the gas error, you should correctly handle all types of gas error, not only the intrinsicGas. Furthermore, I recommend not making a big change to the existing function signature if the goal (awareness of the gas type error) is preserved. For example, `executable` function may have an additional return type of error to see that error type and its small change seems enough.

File: api/api_ethereum.go [hunk 6]
```diff
@@ -610,6 +606,7 @@ func (api *EthereumAPI) Call(ctx context.Context, args EthTransactionArgs, block
 
 	err = blockchain.GetVMerrFromReceiptStatus(status)
 	if err != nil && isReverted(err) && len(result) > 0 {
+		logger.Info("revert error occured", "error", err)
```
Comment (aidan-kwon) [inline api/api_ethereum.go:609]: Revert during Call API is common cases for EN. Is there any reason we print a log here?

File: api/api_public_blockchain.go [hunk 7]
```diff
@@ -604,3 +630,62 @@ func newRPCTransactionFromBlockHash(b *types.Block, hash common.Hash) map[string
 	}
 	return nil
 }
+
+func (args *CallArgs) ToMessage(globalGasCap uint64, baseFee *big.Int, intrinsicGas uint64) (*types.Transaction, error) {
+	// Set sender address or use zero address if none specified.
+	addr := args.From
+
+	// Set default gas & gas price if none were set
+	gas := globalGasCap
```
[the remark this answers is not in the document]
Comment (hqjang-pepper) [reply api/api_public_blockchain.go:639]: If you keep follow the code, and you'll figure out that `args.Gas` is nil. 
So, you should not initialize it with args.gas.
```
args := CallArgs{
To:   &cypressCreditContractAddress,
Data: abiGet,
}
```

[the remark this answers is not in the document]
Comment (hyunsooda) [reply api/api_public_blockchain.go:639]: I couldn't understand what relationship is there.

The line 646 already checks the setting existence of the `globalGasCap`. Thus, the suggested blocks will get rid of the unnecessary code. Please take a look again and would appreciate it if let me know if I'm missing something.

File: api/api_public_blockchain.go [hunk 8]
```diff
@@ -604,3 +630,62 @@ func newRPCTransactionFromBlockHash(b *types.Block, hash common.Hash) map[string
...
+			gasPrice = new(big.Int)
+			if gasFeeCap.BitLen() > 0 || gasTipCap.BitLen() > 0 {
+				gasPrice = math.BigMin(new(big.Int).Add(gasTipCap, baseFee), gasFeeCap)
+			}
+		}
+	}
+	value := new(big.Int)
+	if &args.Value != nil {
+		value = args.Value.ToInt()
+	}
```
[the remark this answers is not in the document]
Comment (hyunsooda) [reply api/api_public_blockchain.go:682]: Make sense, but I was thinking the `args.value` is guaranteed not nil as the existing code does not have such a guard. Could you clarify it?

File: api/api_public_blockchain.go [hunk 9]
```diff
@@ -604,3 +630,62 @@ func newRPCTransactionFromBlockHash(b *types.Block, hash common.Hash) map[string
...
+		// A basefee is provided, necessitating 1559-type execution
+		if &args.GasPrice != nil {
+			// User specified the legacy gas field, convert to 1559 gas typing
+			gasPrice = args.GasPrice.ToInt()
+			gasFeeCap, gasTipCap = gasPrice, gasPrice
+		} else {
+			// User specified 1559 gas fields (or none), use those
+			gasFeeCap = new(big.Int)
+			// Backfill the legacy gasPrice for EVM execution, unless we're all zeros
+			gasPrice = new(big.Int)
```
[the remark this answers is not in the document]
Comment (hyunsooda) [reply api/api_public_blockchain.go:673]: uh uh, I'm sorry.

File: api/api_public_blockchain.go [hunk 10]
```diff
@@ -604,3 +630,62 @@ func newRPCTransactionFromBlockHash(b *types.Block, hash common.Hash) map[string
...
+		gasFeeCap *big.Int
+		gasTipCap *big.Int
+	)
+	if baseFee == nil {
+		// If there's no basefee, then it must be a non-1559 execution
+		gasPrice = new(big.Int)
+		if &args.GasPrice != nil {
+			gasPrice = args.GasPrice.ToInt()
+		}
+		gasFeeCap, gasTipCap = gasPrice, gasPrice
```
[the remark this answers is not in the document]
Comment (hyunsooda) [reply api/api_public_blockchain.go:662]: Let me put the same comment (`I was thinking the args.value is guaranteed not nil as the existing code does not have such a guard. Could you clarify it?`)

File: api/api_public_blockchain.go (hunk 2 above)
Comment (hqjang-pepper) [reply api/api_public_blockchain.go:291]: Yes, it can. It checks all variables and params at args are not nil, or preset.

File: api/api_public_blockchain.go (hunk 8 above)
[the remark this answers is not in the document]
Comment (hqjang-pepper) [reply api/api_public_blockchain.go:682]: You are correct, there was no guard before. However, code that protects in this way is used on the Ethereum side, and I think it is necessary to protect the code once in the `Tomessage` function before converting mgs into a transaction.

File: api/api_public_blockchain.go [hunk 11]
```diff
@@ -604,3 +630,62 @@ func newRPCTransactionFromBlockHash(b *types.Block, hash common.Hash) map[string
...
+			// User specified the legacy gas field, convert to 1559 gas typing
+			gasPrice = args.GasPrice.ToInt()
+			gasFeeCap, gasTipCap = gasPrice, gasPrice
+		} else {
+			// User specified 1559 gas fields (or none), use those
+			gasFeeCap = new(big.Int)
+			// Backfill the legacy gasPrice for EVM execution, unless we're all zeros
+			gasPrice = new(big.Int)
+			if gasFeeCap.BitLen() > 0 || gasTipCap.BitLen() > 0 {
+				gasPrice = math.BigMin(new(big.Int).Add(gasTipCap, baseFee), gasFeeCap)
```
[the remark this answers is not in the document]
Comment (hqjang-pepper) [reply api/api_public_blockchain.go:675]: In fact, this `else` conditional statement is never executed until the dynamic gas fee is implemented. You are right. This part is incomplete and will be corrected when the dynamic gas fee is applied.

File: api/api_public_blockchain.go (hunk 3 above)
Comment (hqjang-pepper) [reply api/api_public_blockchain.go:391]: I put it in in advance before appling the dynamic gas fee.

File: api/api_public_blockchain.go (hunk 4 above)
Comment (hqjang-pepper) [reply api/api_public_blockchain.go:393]: That is out of scope of this PR. I'll apply the error message return during EVM execution later.

File: api/api_ethereum.go (hunk 6 above)
Comment (hqjang-pepper) [reply api/api_ethereum.go:609]: I did it because I thought it would be good to show it as a log. I will remove it again.

File: api/api_public_blockchain.go [hunk 12]
```diff
@@ -604,3 +630,62 @@ func newRPCTransactionFromBlockHash(b *types.Block, hash common.Hash) map[string
...
+		if &args.GasPrice != nil {
+			// User specified the legacy gas field, convert to 1559 gas typing
+			gasPrice = args.GasPrice.ToInt()
+			gasFeeCap, gasTipCap = gasPrice, gasPrice
+		} else {
+			// User specified 1559 gas fields (or none), use those
+			gasFeeCap = new(big.Int)
+			// Backfill the legacy gasPrice for EVM execution, unless we're all zeros
+			gasPrice = new(big.Int)
+			if gasFeeCap.BitLen() > 0 || gasTipCap.BitLen() > 0 {
```
[the remark this answers is not in the document]
Comment (hqjang-pepper) [reply api/api_public_blockchain.go:676]: For now, always false is correct. The reason is that after initializing gasprice and gastipcap, the current kip71 logic is not added, so there are some fields that cannot read values ​​from args. 
Anyway, this logic is not executed before the kip71 hard fork. I will edit the code with a new PR after the hardfork.

File: api/api_public_blockchain.go (hunk 8 above)
[the remark this answers is not in the document]
Comment (hyunsooda) [reply api/api_public_blockchain.go:684]: Seems strange. Then, it means there had no such a guard before. Anyway, seems a good change from now on.

File: api/api_public_blockchain.go (hunk 4 above)
Comment (hyunsooda) [reply api/api_public_blockchain.go:394]: Let me unmark it as resolved for easier tracking.

**Comment by aeharvlee** [comment]
```js
> klay.call({from: rich, to: ca, input: abi, gas: 5000000})
Error: evm: execution reverted
    at web3.js:3278:20
    at web3.js:6805:15
    at web3.js:5216:36
    at <anonymous>:1:1

> eth.call({from: rich, to: ca, input: abi, gas: 5000000})
Error: execution reverted: Caller is not owner
    at web3.js:3278:20
    at web3.js:6805:15
    at web3.js:5216:36
    at <anonymous>:1:1
```

**Comment by hqjang-pepper** [comment]
@aeharvlee 
Thanks for the review.
I only tested with rpc with curl command line tool, so didn't expect the results to be different when testing with ipc.
When testing with rpc, an error appears in the message.
I'll check more.

```
 janghyeong-gyu  ~/workspace/SC_nodetest  curl -H "Content-Type: application/json" --data '{"jsonrpc": "2.0", "method": "klay_call", "params": [{"from": "0x2a4e10255aa7db1b7598347103444004b875525e", "to": "0x96735df8c12282a8fa438688fba74dc228a504b5", "data": "0xa6f9dae10000000000000000000000003e2ac308cd78ac2fe162f9522deb2b56d9da9499"}, "latest"], "id": 78}' http://localhost:8551
{"jsonrpc":"2.0","id":78,"error":{"code":3,"message":"execution reverted: Caller is not owner","data":"0x08c379a00000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000001343616c6c6572206973206e6f74206f776e657200000000000000000000000000"}}
 janghyeong-gyu  ~/workspace/SC_nodetest  curl -H "Content-Type: application/json" --data '{"jsonrpc": "2.0", "method": "eth_call", "params": [{"from": "0x2a4e10255aa7db1b7598347103444004b875525e", "to": "0x96735df8c12282a8fa438688fba74dc228a504b5", "data": "0xa6f9dae10000000000000000000000003e2ac308cd78ac2fe162f9522deb2b56d9da9499"}, "latest"], "id": 78}' http://localhost:8551
{"jsonrpc":"2.0","id":78,"error":{"code":3,"message":"execution reverted: Caller is not owner","data":"0x08c379a00000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000001343616c6c6572206973206e6f74206f776e657200000000000000000000000000"}}
```

```
> klay.call({from:"0x2a4e10255aa7db1b7598347103444004b875525e", to:"0x96735df8c12282a8fa438688fba74dc228a504b5", input:"0xa6f9dae10000000000000000000000003e2ac308cd78ac2fe162f9522deb2b56d9da9499", gas:5000000})
Error: evm: execution reverted
    at web3.js:3278:20
    at web3.js:6805:15
    at web3.js:5216:36
    at <anonymous>:1:1

> eth.call({from:"0x2a4e10255aa7db1b7598347103444004b875525e", to:"0x96735df8c12282a8fa438688fba74dc228a504b5", input:"0xa6f9dae10000000000000000000000003e2ac308cd78ac2fe162f9522deb2b56d9da9499", gas:5000000})
Error: execution reverted: Caller is not owner
    at web3.js:3278:20
    at web3.js:6805:15
    at web3.js:5216:36
    at <anonymous>:1:1

```

**Comment by aeharvlee** [comment]
@hqjang-pepper 
Hmm.... little strange because I got a same situation when I test with curl.

**klay_call**
```bash
➜  klaytn-network-local curl -H "Content-Type: application/json" --data '{"jsonrpc":"2.0","method":"klay_call","params":[{"from": "0xca7a99380131e6c76cfa622396347107aeedca2d", "to": "0x6deae21e44c8f55b6676a1182c8452eccb93a2e0", "input": "0xa6f9dae10000000000000000000000003e2ac308cd78ac2fe162f9522deb2b56d9da9499", "gas": "0x999999"}, "latest"],"id":83}' http://localhost:8551
```

**eth_call**
```bash
➜  klaytn-network-local curl -H "Content-Type: application/json" --data '{"jsonrpc":"2.0","method":"eth_call","params":[{"from": "0xca7a99380131e6c76cfa622396347107aeedca2d", "to": "0x6deae21e44c8f55b6676a1182c8452eccb93a2e0", "input": "0xa6f9dae10000000000000000000000003e2ac308cd78ac2fe162f9522deb2b56d9da9499", "gas": "0x999999"}, "latest"],"id":83}' http://localhost:8551
```

I already checked whether I'm in right branch and HEAD.
Recent commit already applied in my binary🤔 
```
commit 14ed6da840a48b26a69e59603548ba78c394a70e (HEAD -> klay-namespace-revertmessage, pepper/klay-namespace-revertmessage)
Merge: 035c1019 6c829123
Author: hqjang-pepper <pepper.jang@groundx.xyz>
Date:   Thu Jun 9 10:34:14 2022 +0900

    Merge remote-tracking branch 'origin/klay-namespace-revertmessage' into klay-namespace-revertmessage
```

**Comment by hqjang-pepper** [comment]
I changed logic of retriving transaction calldata.

File: api/api_public_blockchain.go [hunk 13]
```diff
@@ -313,12 +296,26 @@ func DoCall(ctx context.Context, b Backend, args CallArgs, blockNrOrHash rpc.Blo
 	// this makes sure resources are cleaned up.
 	defer cancel()
 
-	// Add gas fee to sender for estimating gasLimit/computing cost or calling a function by insufficient balance sender.
-	state.AddBalance(msg.ValidatedSender(), new(big.Int).Mul(new(big.Int).SetUint64(msg.Gas()), msg.GasPrice()))
```
Comment (jimni1222) [inline api/api_public_blockchain.go:317]: Why did you remove this line?

File: api/api_public_blockchain.go [hunk 14]
```diff
@@ -313,12 +296,26 @@ func DoCall(ctx context.Context, b Backend, args CallArgs, blockNrOrHash rpc.Blo
...
+	}
+	msg, err := args.ToMessage(globalGasCap.Uint64(), fixedBaseFee, intrinsicGas)
 	if err != nil {
-		return nil, 0, 0, false, err
+		return nil, 0, 0, 0, err
+	}
+	// The intrinsicGas is checked again later in the blockchain.ApplyMessage function,
+	// but we check in advance here in order to keep StateTransition.TransactionDb method as unchanged as possible
+	// and to clarify error reason correctly to serve eth namespace APIs.
+	// This case is handled by EthDoEstimateGas function.
```
Comment (jimni1222) [inline api/api_public_blockchain.go:312]:
```suggestion
	// This case is handled by DoEstimateGas function.
```

File: api/api_public_blockchain.go [hunk 15]
```diff
@@ -606,3 +645,62 @@ func newRPCTransactionFromBlockHash(b *types.Block, hash common.Hash) map[string
 	}
 	return nil
 }
+
+func (args *CallArgs) ToMessage(globalGasCap uint64, baseFee *big.Int, intrinsicGas uint64) (*types.Transaction, error) {
+	// Set sender address or use zero address if none specified.
+	addr := args.From
```
Comment (jimni1222) [inline api/api_public_blockchain.go:651]: Don't we need to add logic to handle when From is empty?

Comment (jimni1222) [reply api/api_public_blockchain.go:651]: Never mind.

File: api/tx_args.go [hunk 16]
```diff
@@ -445,3 +447,43 @@ func (args *AccountUpdateTxArgs) toTransaction() (*types.Transaction, error) {
 
 	return tx, nil
 }
+
+// isReverted checks given error is vm.ErrExecutionReverted
```
Comment (jimni1222) [inline api/tx_args.go:451]: Why did you decide to move those logic to here?
How about in `api_public_blockchain.go` file?

File: api/api_public_blockchain.go [hunk 17]
```diff
@@ -606,3 +645,62 @@ func newRPCTransactionFromBlockHash(b *types.Block, hash common.Hash) map[string
...
+		gasFeeCap, gasTipCap = gasPrice, gasPrice
+	} else {
+		// A basefee is provided, necessitating 1559-type execution
+		if &args.GasPrice != nil {
+			// User specified the legacy gas field, convert to 1559 gas typing
+			gasPrice = args.GasPrice.ToInt()
+			gasFeeCap, gasTipCap = gasPrice, gasPrice
+		} else {
+			// User specified 1559 gas fields (or none), use those
+			gasFeeCap = new(big.Int)
```
[the remark this answers is not in the document]
Comment (hqjang-pepper) [reply api/api_public_blockchain.go:686]: No problem. I'll add it.

File: api/api_ethereum.go [hunk 18]
```diff
@@ -1577,7 +1577,7 @@ func EthDoEstimateGas(ctx context.Context, b Backend, args EthTransactionArgs, b
 	// Binary search the gas requirement, as it may be higher than the amount used
 	var (
 		lo  uint64 = params.TxGas - 1
-		hi  uint64
+		hi  uint64 = 99999999
```
Comment (JayChoi1736) [inline api/api_ethereum.go:1580]: could you explain what does this number mean?

Comment (hqjang-pepper) [reply api/api_ethereum.go:1580]: Oh, it's just a enough big number for highest gas limit. Never mind.

File: api/api_public_blockchain.go (hunk 13 above)
Comment (hqjang-pepper) [reply api/api_public_blockchain.go:317]: It was my misunderstood. I didn't fully understand the role of `EthStateOverride` in `api_ethereum.go`at that time.
I'll restore this line.

File: api/api_ethereum.go (hunk 18 above)
Comment (kjeom) [reply api/api_ethereum.go:1580]: @hqjang-pepper How about changing it to params.upperGasLimit or math.MaxUint64?

Comment (hqjang-pepper) [reply api/api_ethereum.go:1580]: upperGasLimit seems good idea. Thanks

Comment (JayChoi1736) [reply api/api_ethereum.go:1580]: Oh, what i wondered is that it looks like hi doesn't need default value. it seems hi are always initialized as  uinit64(*args.Gas) or params.UpperGasLimit

File: api/api_public_blockchain.go [hunk 19]
```diff
@@ -359,7 +370,11 @@ func (s *PublicBlockChainAPI) EstimateComputationCost(ctx context.Context, args
 
 // EstimateGas returns an estimate of the amount of gas needed to execute the given transaction against the latest block.
 func (s *PublicBlockChainAPI) EstimateGas(ctx context.Context, args CallArgs) (hexutil.Uint64, error) {
-	return s.DoEstimateGas(ctx, s.b, args, s.b.RPCGasCap())
+	gasCap := uint64(0)
+	if rpcGasCap := s.b.RPCGasCap(); rpcGasCap != nil {
+		gasCap = rpcGasCap.Uint64()
+	}
+	return s.DoEstimateGas(ctx, s.b, args, big.NewInt(int64(gasCap)))
 }
```
Comment (henry-will) [inline api/api_public_blockchain.go:378]: Are 350 ~ 353 lines same as this? If so, I think upper case better.  What do you think?

**Review by jimni1222 — APPROVED** [review]
LGTM =) I tested few cases and confirmed that it works! 

```
> eth.call({from: executor, to: ca, input: abi, gas: 5000000})
> eth.call({from: executor, to: ca, data: abi, gas: 5000000})
> klay.call({from: executor, to: ca, input: abi, gas: 5000000})
> klay.call({from: executor, to: ca, data: abi, gas: 5000000})
> eth.estimateGas({from: executor, to: ca, input: abi, gas: 5000000})
> eth.estimateGas({from: executor, to: ca, data: abi, gas: 5000000})
> klay.estimateGas({from: executor, to: ca, input: abi, gas: 5000000})
> klay.estimateGas({from: executor, to: ca, data: abi, gas: 5000000}) Error: execution reverted: Caller is not owner
    at web3.js:3278:20
    at web3.js:6805:15
    at web3.js:5216:36
    at <anonymous>:1:1
```

```
> eth.call({from: deployer, to: ca, input: abi, gas: 5000000})
> eth.call({from: deployer, to: ca, data: abi, gas: 5000000})
> klay.call({from: deployer, to: ca, input: abi, gas: 5000000})
> klay.call({from: deployer, to: ca, data: abi, gas: 5000000}) "0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000" 
```

```
> eth.estimateGas({from: deployer, to: ca, data: abi, gas: 5000000})
> eth.estimateGas({from: deployer, to: ca, input: abi, gas: 5000000})
> klay.estimateGas({from: deployer, to: ca, data: abi, gas: 5000000})
> klay.estimateGas({from: deployer, to: ca, input: abi, gas: 5000000})
62951
```
